### DISCUSSION TITLE
Keyboard on remote not working

### DISCUSSION THREAD

**Discussion by johnh530** [rustdesk/rustdesk#2828] [Q&A]
We have installed rustdesk on a mac, windows-7 and windows-11 server.

The mac version is 1.1.9

We are using the windows computers as the remote computers.

Everything seems to be working ok except for the keyboard.

At one time I was able to access the windows-11 server with keyboard functionality after not being able to.

The windows computers may or not have a keyboard attached.

Is it necessary to have a keyboard attached to the windows computers to be able to have the local computer's keyboard function as the remote computer's keyboard?

Any other issues with using a mac (like having multiple sessions going) which cause issues for keyboard.

Permissions seem to be correct as mouse movement and clicks are working as expected.

johnh...

**Comment by rustdesk** [comment]
try out nightly build

**Reply by johnh530** [reply]
Well I finally got all the pieces needed to build and I ran ./target/build/rustdesk

It opened ok but it produced :-{
[2023-01-20T23:20:43Z ERROR librustdesk::keyboard] rdev Error: EventTapError

Not sure I had the latest nightly build - I followed the directions at:
https://rustdesk.com/docs/en/dev/build/osx/
====
cd
git clone https://github.com/microsoft/vcpkg
cd vcpkg
git checkout 2022.04.12
./bootstrap-vcpkg.sh
brew install nasm yasm
./vcpkg install libvpx libyuv opus
git clone https://github.com/rustdesk/rustdesk
cd rustdesk
export VCPKG_ROOT=$HOME/vcpkg
wget https://github.com/c-smile/sciter-sdk/raw/master/bin.osx/libsciter.dylib
cargo run
====
Then ran
./target/build/rustdesk 

I verified that a PC could remote in to my mac without issues.

johnh...

**Reply by johnh530** [reply]
Ok - I now see there are Nigthly Releases - found the dmg for 1.2.0 and was able to run that.
Now I get clicking feedback (little poping sounds) with 1.2.0 - the 1.1.9 version give no feedback.
Still no joy.

**Reply by johnh530** [reply]
1.1.8 seems to work.
Would be nice to have this working on newer version.
johnh...
